Einarbeitung der Änderungsvorschläge (#1)
jo2/Masterarbeit: Expose/expose.tex

\maketitle

\section{Motivation}
Laut \cite{Balzert2009} entfallen 80\% des Aufwands in der Softwareentwicklung auf deren Wartung, wobei 40\% davon benötigt werden, um die zu wartende Software zu verstehen.
Ein erheblicher Teil der Kosten für die Softwareentwicklung entfällt damit auf das Verstehen der Software.
Wenn insgesamt die Entwicklungskosten reduziert werden sollen, stellt dieser Umstand einen wesentlichen Hebel dar.

> Denn der Wartungsaufwand im Übrigen ist vielfach kaum zu verringern, weil sich die Anforderungen an die Software im Laufe der Zeit ändern, weil auftretende Probleme behoben werden müssen oder weil Abhängigkeiten auf andere Software, Frameworks oder Komponenten angepasst werden müssen.

Reviewer: ```suggestion
Denn der Wartungsaufwand im Übrigen ist vielfach kaum zu verringern, weil sich die Anforderungen an die Software im Laufe der Zeit ändern, weil auftretende Probleme behoben werden müssen oder weil Abhängigkeiten zu anderer Software, Frameworks oder Komponenten angepasst werden müssen.
```

Software, die diese Anforderungen erfüllt, beinhaltet Code, der nach bestimmten Richtlinien gebildet wird.
Zu diesen Regeln gehört beispielsweise, dass Code auf einer Ebene einheitlich eingerückt wird, dass öffnende geschweifte Klammern in der gleichen Zeile stehen wie der Code, der die Klammer erfordert, dass Variablennamen sprechend sind und nicht nur aus einem Buchstaben bestehen oder dass Parameter auf ihre Gültigkeit überprüft werden, bevor mit ihnen gearbeitet wird.

Die Einhaltung solcher Regeln scheint auf den ersten Blick nur von geringer Bedeutung zu sein, allerdings lenken beispielsweise Unstimmigkeiten in der Formatierung beim Lesen des Codes ab und sie erschweren es, sich auf den eigentlichen Code zu konzentrieren.
In \cite{PJ2015} und \cite{SP2011} werden diese Formatverstöße als Hintergrundrauschen beschrieben, das vom eigentlichen Code ablenkt. 

\section{Problemstellung}
Code, der nach den oben beschriebenen Regeln geschrieben wurde, ist laut \cite{PJ2015} leichter verständlich und dadurch auch leichter erlernbar und veränderbar.

> Um die Teammitglieder dazu zu bringen, sich an die vereinbarten Regeln zu halten, gibt es zwei Möglichkeiten.

Reviewer: ```suggestion
Um die Mitglieder eines Projektteams dazu zu bringen, sich an die vereinbarten Regeln zu halten, gibt es zwei Möglichkeiten.
```

> Zum einen kann im Reviewprozess des Projekts ein automatischer Test eingebaut werden, der dafür sorgt, dass grundsätzlich funktionierender Code dann nicht angenommen wird, wenn er gegen festgelegte Regeln verstößt, wenn er also beispielsweise Stylingrichtlinien nicht einhält oder es potentiell unsichere Variablenzugriffe gibt.

Reviewer: ```suggestion
Zum einen kann im Reviewprozess des Projekts ein automatischer Test eingebaut werden, der dafür sorgt, dass grundsätzlich funktionierender Code dann nicht angenommen wird, wenn er gegen festgelegte Regeln verstößt, wenn er also beispielsweise Stylingrichtlinien nicht einhält oder es potentiell unsichere Variablenzugriffe gibt. Nachteil dieser Vorgehensweise ist eine gewisse Frustration der Teammitglieder, weil Code zurückgewiesen wird???
```

Zum anderen können die Teammitglieder durch Belohnungen dazu motiviert werden, von sich aus die festgelegten Regeln einzuhalten und dies auch selbst zu überprüfen. 

> Es stellt sich nun die Frage, ob eine solche Motivation in durch Gamification erreicht werden kann.

Reviewer: ```suggestion
Es stellt sich nun die Frage, ob eine solche Motivation durch Gamification erreicht werden kann.
```

Bei Gamification werden Methoden genutzt, die aus Computerspielen bekannt sind.
Dazu gehören beispielsweise Levels, Badges oder Leader Boards, die das Team oder ein einzelnes Teammitglied erreichen kann. 

> Das Ablehnen von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.

Reviewer: dass sich dies auf das oben bezieht, wird nicht so ganz deutlich. Vielleicht:
```suggestion
Die oben beschriebene Ablehnung von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.
```
oder
```suggestion
Das Ablehnen von eingereichtem Code auf Grund von Verstößen gegen Richtlinien, wie es oben beschrieben wurde, führt dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.
```
oder findest du, dass der Bezug auch so klar wird?

> Die oben beschriebene Ablehnung von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.

Reviewer: ```suggestion
Die oben beschriebene Ablehnung von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt zwar dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.
```

Reviewer: Das zwar muss weg!!! Schrieb ich doch gestern schon?! Da ist kein korrespondierendes aber

Es ergibt sich also ein Lerneffekt, der bewirkt, dass die Teilnehmer auch in Zukunft qualitativ hochwertigen Code einreichen.

> Im Rahmen dieser Arbeit soll untersucht werden, ob beziehungsweise in wie weit auszuwählende Gamification-Elemente ebenfalls die Motivation erhöhen, qualitativ hochwertigeren Code einzureichen.

Reviewer: ```suggestion
Im Rahmen dieser Arbeit soll untersucht werden, ob beziehungsweise in wie weit auszuwählende Gamification-Elemente ebenso die Motivation erhöhen, qualitativ hochwertigeren Code einzureichen, und daneben 
```

Reviewer: ```suggestion
Im Rahmen dieser Arbeit soll untersucht werden, ob beziehungsweise in wie weit auszuwählende Gamification-Elemente ebenfalls die Motivation erhöhen, qualitativ hochwertigeren Code einzureichen, und darüber hinaus 
```

> Neben qualitativ hochwertigem Code könnte so durch einen gewissen \glqq Spaßfaktor\grqq{} auch eine höhere Zufriedenheit der Teammitglieder erreicht werden.

Reviewer: ```suggestion
durch einen gewissen \glqq Spaßfaktor\grqq{} sogar noch eine höhere Zufriedenheit der Teammitglieder erreicht werden kann.
```

> Neben qualitativ hochwertigerem Code könnte so durch einen gewissen "Spaßfaktor" auch eine höhere Zufriedenheit der Teammitglieder und somit eventuell auch eine intrinsische Motivation zum Schreiben von qualitativ hochwertigerem Code erreicht werden.

Reviewer: 2 /3 mal qualtitativ hochwertigeren Code finde ich eigenartig. vielleicht hier nur
Neben qualitativ hochwertigem Code könnte so....
und das mit der intrinsischen Motivation zu hochwertigerem Code würde ich ganz weglassen.
```suggestion
Neben qualitativ hochwertigem Code könnte so durch einen gewissen "Spaßfaktor" auch eine höhere Zufriedenheit der Teammitglieder erreicht werden.
```

Reviewer: warum macht der Spaßfaktor rot, sollte mir das zu denken geben?

Anschließend wird eine Balancierungsmatrix für eine gleichmäßige Verteilung der Punkte, auf denen Levels, Leader Boards und ähnliches basieren, erarbeitet.
Abschließend wird ein Feldversuch durchgeführt, um festzustellen, ob und in wie weit die vorgestellten Maßnahmen die Softwarequalität nachhaltig verbessern konnten.

\section{Vorgehensweise}
Zunächst werden die Grundlagen dieser Arbeit vorgestellt.
Es wird erläutert, was Gamification ist und welche Gamification-Elemente in dieser Arbeit Verwendung finden werden.
Daneben wird im Rahmen der Vorstellung verwandter Arbeiten darauf eingegangen, welche Implementierungen und Implementierungsansätze es bereits für Gamification-Elemente in Zusammenhang mit der Code-Qualität gibt.
Außerdem wird die Codeanalyseplattform coderadar vorgestellt, die als Basis für die im Rahmen dieser Arbeit erfolgende Implementierung dienen wird.

Im Anschluss daran wird die Einbindung von Gamification-Elementen beschrieben.
Dazu gehört zunächst die Auswahl der Metriken, anhand derer die Codequalität bewertet werden soll.
Entsprechend diesen Metriken wird der konkrete Code dann untersucht.

> Anhand des Ergebnisses dieser Untersuchung werden anschließend die Punkte verteilt, auf denen später Levels oder Leader Boards beruhen.

Reviewer: ```suggestion
Entsprechend dem Ergebnis dieser Untersuchung werden anschließend die Punkte verteilt, auf denen später Levels oder Leader Boards beruhen.
```

Auf Basis dieser Anforderungen werden dann die erforderlichen Änderungen an der coderadar-Plattform modelliert.
Es folgt die eigentliche Implementierung.

Anschließend wird die Evaluierung durchgeführt.
Dazu wird zunächst ein Projekt zur Untersuchung in coderadar eingecheckt, und anhand der ausgewählten Metriken wird dessen Code-Qualität bewertet.
Der hier ermittelte Wert soll als Grundlage für die Messung der künftigen Bewertungsänderungen dienen.
Über eine gewisse Zeit wird das Projekt dann im Rahmen eines Feldversuchs verfolgt und es wird anhand der ausgewählten Metriken ermittelt, in wie weit sich die Bewertung des untersuchten Projektes verändert.
Zusätzlich dazu sollen die Projektteammitglieder zu ihrer Arbeit mit den Gamification-Elementen in coderadar befragt werden.

Abschließend wird das Ergebnis der Arbeit zusammengefasst.
Als Grundlage für dieses Fazit dienen dabei zum einen die Beobachtungen über die Veränderung der Code-Qualität und zum anderen die Ergebnisse aus der Befragung der Projektteammitglieder.
Außerdem wird reflektiert, welche Herausforderungen es bei der Implementierung und bei der Durchführung des Feldversuchs gab.

> Zum Schluss wird ein Ausblick auf die weitere Arbeit gegeben.

Reviewer: auf welche weitere Arbeit???

> Außerdem wird reflektiert, welche Herausforderungen es bei der Implementierung und der Durchführung des Feldversuchs gab.

Reviewer: ich würde ein zweites bei setzen, weil es zwei ganz verschiedene Sachen sind
```suggestion
Außerdem wird reflektiert, welche Herausforderungen es bei der Implementierung und bei der Durchführung des Feldversuchs gab.
```

\maketitle

\section{Motivation}

> Laut \cite{Balzert2009} fallen 80\% des Aufwands in der Softwareentwicklung während der Wartung an, wobei 40\% davon zum Verstehen der Software benötigt werden.

Reviewer: [Bal09] geht davon aus, dass 80 % des Aufwands in der Softwareentwicklung auf die Wartung und hier wiederum 40 % auf das Verstehen der Software entfällt.

> Laut \cite{Balzert2009} entfallen 80\% des Aufwands in der Softwareentwicklung auf deren Wartung wobei 40\% davon benötigt werden, um die zu wartende Software zu verstehen.

Reviewer: ```suggestion
Laut \cite{Balzert2009} entfallen 80\% des Aufwands in der Softwareentwicklung auf deren Wartung, wobei 40\% davon benötigt werden, um die zu wartende Software zu verstehen.
```

Author: Danke

> Laut \cite{Balzert2009} entfallen 80\% des Aufwands in der Softwareentwicklung auf deren Wartung wobei 40\% davon benötigt werden, um die zu wartende Software zu verstehen.

Reviewer: [Bal09] geht davon aus, dass 80 % des Aufwands in der Softwareentwicklung auf die Wartung und hier wiederum 40 % auf das Verstehen der Software entfällt.

> Laut \cite{Balzert2009} entfallen 80\% des Aufwands in der Softwareentwicklung auf deren Wartung wobei 40\% davon benötigt werden, um die zu wartende Software zu verstehen.

Reviewer: ```suggestion
\cite{Balzert2009} geht davon aus, dass 80 % des Aufwands in der Softwareentwicklung auf die Wartung und hier wiederum 40 % auf das Verstehen der Software entfällt.
```

> Somit liegt auch ein großer Teil der Kosten im Bereich der Wartung.

Reviewer: ```suggestion
Zwangsläufig stellt somit das Verstehen der Software einen erheblichen Teil der Kosten der Softwareentwicklung dar. 
```

> Der Wartungsaufwand an sich lässt sich selten reduzieren, da weiterhin Anforderungsänderungen eingereicht werden können, Problembehebungen notwendig werden können oder Abhängigkeiten aktualisiert werden müssen.

Reviewer: ```suggestion
Wenn insgesamt die Entwicklungskosten reduziert werden sollen, stellt dieser Umstand einen wesentlichen Hebel dar. Denn der Wartungsaufwand im Übrigen lässt sich vielfach nicht reduzieren, weil sich die Anforderungen an die Software im Laufe der Zeit ändern, weil auftretende Probleme behoben werden müssen oder weil Abhängigkeiten mit anderer Software oder Komponenten angepasst werden müssen.
```

> Allerdings lässt sich der Aufwand für das Verstehen des Codes reduzieren.

Reviewer: ```suggestion
Aufwand für das Verstehen der Software lässt sich bereits dadurch reduzieren, dass der Code gewissen Qualitätsstandards entspricht, so dass das Verstehen der Software weniger aufwändig ist.
```

Dazu ist es notwendig, dass der Code gewisse Qualitätsstandards erfüllt.

Ein Aspekt von qualitativ hochwertiger Software ist laut \cite{ISO-25010} eine hohe Qualität in Betrieb und Wartung.

> Dazu gehören die Analysierbarkeit, Erlernbarkeit, Veränderbarkeit, Stabilität und Wartbarkeit der Software.

Reviewer: ```suggestion
Qualitativ hochwertige Software zeichnet sich nach [ISO11] dadurch aus, dass sie gut gewartet, analysiert, erlernt und verändert werden kann sowie stabil ist.
```

> Ein Aspekt von qualitativ hochwertiger Software ist laut \cite{ISO-25010} eine hohe Qualität in Betrieb und Wartung.
> Dazu gehören die Analysierbarkeit, Erlernbarkeit, Veränderbarkeit, Stabilität und Wartbarkeit der Software.

Author: ```suggestion
Qualitativ hochwertige Software zeichnet sich nach \cite{ISO-25010} dadurch aus, dass sie stabil ist und gut gewartet, analysiert, erlernt und verändert werden kann.
```

Um Code zu schreiben, der diese Anforderungen erfüllt, gibt es Richtlinien, an die sich alle Entwickler in einem Team halten sollten.

> Diese Richtlinien besitzen zwar keine Allgemeingültigkeit, aber sie werden meist zu Projektbeginn festgelegt und gelten dann für die gesamte Projektlaufzeit.

Reviewer: ```suggestion
Eine Software, die diese Anforderungen erfüllt, beinhaltet einen Code, der nach bestimmten Richtlinien gebildet wird. Voraussetzung ist jedoch, dass sich alle Entwicklern eines Teams auch an diese vorher festgelegten Richtlinien halten.
```

Author: Es sieht aus, als wäre diese Anmelrkung in der falschen Zeile gelandet, kann das?

Reviewer: Stephan hatte das von mir üüberarbeitete (und von dir bereits hier eingefügte und überarbeitete) nicht gekannt, als er seinen Vorschlag machte.

Reviewer: Ich denke diese Anmerkung (die neueste /unterste 42) fasst Zeile 41 und 42 zu einer Zeile zusammen - das kannst du aber am besten selbst richtig machen

Reviewer: und bei Entwicklern muss noch das n weg

> Voraussetzung ist jedoch, dass sich alle Entwickler eines Teams an diese vorher festgelegten und für die gesamte Projektlaufzeit geltenden Richtlinien halten.

Reviewer: der Satz ist hier störend und überflüssig, das mit der Voraussetzung kommt auch scon zu beginn des nächsten Absazes

> Zu diesen Regeln gehört beispielsweise, dass Code auf einer Ebene einheitlich eingerückt wird, dass öffnende geschweifte Klammern in der gleichen Zeile stehen wie der Code, der die Klammer erfordert oder aber auch dass Variablennamen sprechend sind und nicht aus einem Buchstaben bestehen, oder dass Parameter auf ihre Gültigkeit überprüft werden, bevor mit ihnen gearbeitet wird. 

Reviewer: ```suggestion
Zu diesen Regeln gehört beispielsweise, dass der Code auf einer Ebene einheitlich eingerückt wird, dass öffnende geschweifte Klammern in der gleichen Zeile sind, wie der Code, der die Klammer erfordert, dass Variablennamen sprechend sind und nicht nur aus einem Buchstaben bestehen oder dass Parameter auf ihre Gültigkeit überprüft werden, bevor mit ihnen gearbeitet wird.
```

Die Einhaltung solcher Regeln scheint auf den ersten Blick nur von geringer Bedeutung zu sein, allerdings lenken beispielsweise Unstimmigkeiten in der Formatierung beim Lesen des Codes ab und sie erschweren es, sich auf den eigentlichen Code zu konzentrieren.

> In \cite{PJ2015} und \cite{SP2011} werden diese Formatverstöße als Hintergrundrauschen beschrieben, das vom eigentlichen Code ablenken. 

Author: ```suggestion
In \cite{PJ2015} und \cite{SP2011} werden diese Formatverstöße als Hintergrundrauschen beschrieben, das vom eigentlichen Code ablenkt. 
```

\section{Problemstellung}
Code, der nach den oben beschriebenen Regeln geschrieben wurde, ist laut \cite{PJ2015} leichter verständlich und dadurch auch leichter erlernbar und veränderbar.
Um die Teammitglieder dazu zu bringen, sich an diese Regeln zu halten, gibt es zwei Möglichkeiten.

> Zum einen kann im Reviewprozess des Projekts ein automatischer Test eingebaut werden, der dafür sorgt, dass eigentlich funktionierender Code nicht angenommen wird, wenn er beispielsweise gegen Stylingrichtlinien verstößt oder es potentiell unsichere Variablenzugriffe gibt.

Reviewer: ```suggestion
Zum einen kann im Reviewprozess des Projekts ein automatischer Test eingebaut werden, der dafür sorgt, dass grundsätzlich funktionierender Code dann nicht angenommen wird, wenn er gegen festlegte Regeln verstößt, wenn er also beispielsweise gegen Stylingrichtlinien verstößt oder es potentiell unsichere Variablenzugriffe gibt.
```

> Zum anderen können die Teammitglieder durch Belohnungen dazu motiviert werden, selber für die Einhaltung der Regeln zu sorgen und dies zu überprüfen.

Reviewer: ```suggestion
Zum anderen können die Teammitglieder durch Belohnungen dazu motiviert werden, von sich aus dafür zu sorgen, dass sie die festgelegten Regeln einhalten und dies auch selbst überprüfen. 
```

Um eine solche Motivation zu erreichen, kann Gamification verwendet werden.
Hierbei werden Methoden genutzt, die aus Computerspielen bekannt sind.

> Dazu gehören beispielsweise Elemente wie Levels, Badges oder Leader Boards. 

Reviewer: ```suggestion
Hierzu gehören beispielsweise Levels, Badges oder Leader Boards, die das Team oder das einzelne Teammitglied erreichen kann. 
```

> Um eine solche Motivation zu erreichen, kann Gamification verwendet werden.

Reviewer: In dieser Arbeit soll untersucht werden,  ob eine solche Motivation durch die Verwendung von Gamification erreicht werden kann./erhöht werden kann.

Reviewer: nein das kommt ja erst unten

> Um eine solche Motivation zu erreichen, kann Gamification verwendet werden.

Reviewer: der kommentar davor hat sich erledigt, ich kann ihn nur nicht löschen
```suggestion
Um eine solche Motivation zu erreichen, könnte Gamification verwendet werden.
```
Was hältst du von könnte, um den Unsicherheitsfaktor, den du untersuchen willst, mehr zu betonen?

Reviewer: oder so wie ich es unten zu 52+53 schreibe

> Hierbei werden Methoden genutzt, die aus Computerspielen bekannt sind.

Reviewer: Vielleicht:
Es stellt sich die Frage, ob eine solche Motivation durch Gamification erreicht werden kann./erhöht werden kann.
Bei der(oder ohne Artikel?? vgl. Code)  Gamification werden Methoden genutzt, die aus Computerspielen...

> Das Ablehnen von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.

Reviewer: Dieser Satz müsste jetzt nach oben hinter Satz 50
Statt "Das Ablehnen" ist mE "Die Ablehnung von eingereichtem Code... "dann besser

Hast du eine QuelleLiteraturangabe hierzu?

> Das Ablehnen von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.

Reviewer: gut wäre, da noch eine quelle Literaturangabe für zu haben

Author: Eine wirkliche Literaturquelle ist hier denke ich nicht so einfach zu finden, weil meiner Meinung nach eher banal ist. An der Stelle würde ich eher auf eigene Erfahrung zurückgreifen.

Es ergibt sich also ein Lerneffekt, der bewirkt, dass die Teilnehmer auch in Zukunft qualitativ hochwertigen Code einreichen.

> Im Rahmen dieser Arbeit soll untersucht werden, in wie weit die ausgewählten Gamification-Elementen zu einer extrinsischen Motivation und somit qualitativ hochwertigem Code beitragen.

Reviewer: in der Pädagogik ist extrinsische Motivation Motivation die von außen kommt (Strafenvermeidung, Belohnungen durch den Lehrer) , pädagogisch wertvoller ist es, beim Schüler intrinsische Motivation zu erzeugen

Reviewer: intrinsisch ist motivation aus dem eigenen Inneren heraus...
ist Gamification intrinsisch oder extrinsisch oder erst extrinsisch, dann entsteht daraus intrinsische?

Reviewer: vielleicht lässt du das mit extrinsisch oder intrinsisch weg oder ist das irgendwofür wichtig?

Reviewer: dsa mit der Unterscheidung zwischen extrinsisch und intrinsisch meine ich

Reviewer: heute morgen: frisch ausgeruht:
oder so, wie ich es unten vorschlage

> Im Rahmen dieser Arbeit soll untersucht werden, in wie weit die ausgewählten Gamification-Elementen zu einer extrinsischen Motivation und somit qualitativ hochwertigem Code beitragen.

Reviewer: ```suggestion
Im Rahmen dieser Arbeit soll untersucht werden, in wie weit die ausgewählten Gamification-Elemente die Motivation erhöhen, qualitativ hochwertigeren Code einzureichen.
```
einzureichen steht im Satz davor,. 
hochwertigeren Code zu entwickeln / zu schreiben /....

> Im Rahmen dieser Arbeit soll untersucht werden, in wie weit die ausgewählten Gamification-Elementen zu einer extrinsischen Motivation und somit qualitativ hochwertigem Code beitragen.

Reviewer: ```suggestion
Im Rahmen dieser Arbeit soll untersucht werden, ob beziehungsweise in wie weit auszuwählende Gamification-Elemente ebenfalls die Motivation, sich an die vereinbarten Regeln zu halten, verstärken. (oder erhöhen)
 oder vielleicht besser, da du oben vorher nicht von motivation schreibst
....auszuwählende Gamification-Elemente die Motivation, sich an die vereinbarten Regeln zu halten, extrinsisch verstärken.
Neben qualitativ hochwertigerem Code könnte so durch einen gewissen "Spaßfaktor" auch eine höhere Zufriedenheit der Mitarbeiter /Teammitglieder erreicht werden.
```
oder wie sagt man das bei euch mit der Mitarbeiterzufriedenheit Arbeitsfreude...oder so

Reviewer: oh je!
```suggestion
Im Rahmen dieser Arbeit soll untersucht werden, in wie weit auszuwählende Gamification-Elemente die Motivation, sich an die vereinbarten Regeln zu halten, extrinsich verstärken können.
```
danach käme jetzt 61: Neben qualitativ hochwertigem Code...
was hältst dudavon? oder kommt das mit dem Spaßfaktor in dieser Seminararbeit noch nicht??

Reviewer: zum Thema "Ausgewählte" "auszuwählende"... Gamification-Elem vgl. den Kommentar nach 72

Reviewer: hier meine ich das neue 61 von weiter oben, nichts das alte von unten

\section{Zielsetzung}
Ziel dieser Arbeit ist die Erweiterung der Codeanalyseplattform coderadar um Gamification-Elemente zur Steigerung der Code-Qualität.

> Dazu werden zunächst ausgewählte Gamification-Elemente in coderadar implementiert.

Reviewer: zu ausgewählte vgl. den KOmmentar nach 71

Anschließend wird eine Balancierungsmatrix für eine gleichmäßige Verteilung der Punkte, auf denen Levels, Leader Boards und ähnliches basieren, erarbeitet.

> Abschließend wird ein Feldversuch durchgeführt, um festzustellen, in wie weit die vorgestellten Maßnahmen die Softwarequalität nachhaltig verbessern konnten.

Reviewer: Wäre besser:
```suggestion
Abschließend wird ein Feldversuch durchgeführt, um festzustellen, ob und in wie weit die vorgestellten Maßnahmen die Softwarequalität nachhaltig verbessern konnten.
```
??

Dazu werden zunächst ausgewählte Gamification-Elemente in coderadar implementiert.
Anschließend wird eine Balancierungsmatrix für eine gleichmäßige Verteilung der Punkte, auf denen Levels, Leader Boards und ähnliches basieren, erarbeitet.
Abschließend wird ein Feldversuch durchgeführt, um festzustellen, in wie weit die vorgestellten Maßnahmen die Softwarequalität nachhaltig verbessern konnten.

\section{Vorgehensweise}
Zunächst werden die Grundlagen dieser Arbeit vorgestellt.
Dazu gehört die Erläuterung, was Gamification ist und in wie weit Gamification-Elemente in dieser Arbeit Verwendung finden.
Daneben wird im Rahmen der Vorstellung verwandter Arbeiten darauf eingegangen, welche Implementierungen und Implementierungsansätze es bereits für Gamification-Elemente in Zusammenhang mit der Code-Qualität gibt.
Außerdem wird die Codeanalyseplattform coderadar vorgestellt, die als Basis für die im Rahmen dieser Arbeit erfolgende Implementierung dienen wird.

> Anschließend wird das Spielkonzept vorgestellt.

Author: ```suggestion
Im Anschluss daran wird das Spielkonzept vorgestellt.
```

> Dazu gehört zunächst die Auswahl der Metriken, anhand derer die Codequalität bewertet.

Author: ```suggestion
Dazu gehört zunächst die Auswahl der Metriken, anhand derer die Codequalität bewertet wird.
```

> Im Anschluss daran wird die Einbindung von Gamification-Elementen erläutert.

Reviewer: wann wählst du die gamification-elemente aus (oben schrieb ich "die auszuwählenden..." und du die ausgewählten
oder ist es oben sinnvoller statt die ausgewählten / ausgewählte / auszuwählende..Gamification-Elemente
einzelne Gamific-E oder verschiedene Gamific-E zu schreiben?

Author: oben ist denke ich auszuwählende ganz gut. Die entsprechenden Elemente sollen im Kapitel nach dem Grundlagenkapitel vrgestellt werden.

Entsprechend diesen Metriken wird der konkrete Code dann untersucht.
Anhand des Ergebnisses dieser Untersuchung werden anschließend die Punkte verteilt, auf denen später Levels oder Leader Boards beruhen.
Des Weiteren werden die Spielmechanismen, die Spielregeln und das Spielziel erläutert.
Außerdem wird eine Balancierungsmatrix ausgearbeitet, die für eine gleichmäßig gewichtete Punkteverteilung für unterschiedliche Aktionen sorgt.

Danach wird auf die Implementierung eingegangen.
Zunächst wird die Modellierung der erforderlichen Änderungen an der coderadar-Plattform vorgenommen.
Auf Basis dieser Modellierung wird eine Herangehensweise ausgearbeitet.
Dann erfolgt die eigentliche Implementierung.

Anschließend erfolgt die Evaluierung.

> Dazu wird zunächst ein Projekt zur Untersuchung in coderadar eingecheckt und anhand der ausgewählten Metriken bewertet.

Author: ```suggestion
Dazu wird zunächst ein Projekt zur Untersuchung in coderadar eingecheckt und anhand der ausgewählten Metriken wird die Code-Qulaität bewertet.
```

Diese Bewertung soll als Basis für den Feldversuch dienen.
Über eine gewisse Zeit wird das Projekt dann verfolgt und es wird beobachtet, in wie weit sich die Bewertung des untersuchten Projektes verändert.

> Anschließend sollen die Projektteammitglieder hinsichtlich der Effizienz der Gamification-Elemente in coderadar befragt werden.

Author: ```suggestion
Zusätzlich dazu sollen die Projektteammitglieder hinsichtlich der Effizienz der Gamification-Elemente in coderadar befragt werden.
```

> Des Weiteren werden die Spielmechanismen, die Spielregeln und das Spielziel beschrieben.

Reviewer: der Satz steht an dieser Stelle falsch!! Wo muss er jetzt nach den Änderungen hin?

Reviewer: oder ist nur Satz 79 jetzt überflüssig geworden (steckt der Inhalt nicht schon in 76 drin??
oder satz 79 hinter 76, dann 77 aber den Satz nicht mit diese einleiten, sondern eher "Die Balancierungsmatrix...").  oder ist dieser Satz überflüssig?   Danach dann78

Author: habe es angepasst.

Reviewer: Ist das inhaltlich richtet: Punkte gleichmäßig gewichtet verteilen  ??

Abschließend wird das Ergebnis der Arbeit zusammengefasst.

> Auf Basis der Bewertungsveränderung und der Ergebnisse der Umfrage wird ein Fazit zu der vorgestellten Implementierung von Gamification-Elementen und ihrem Einfluss auf nachhaltige Code-Qualität gezogen.

Author: ```suggestion
Auf Basis der Bewertungsveränderung und der Ergebnisse der Umfrage wird ein Fazit zur vorgestellten Implementierung von Gamification-Elementen und ihrem Einfluss auf nachhaltige Code-Qualität gezogen.
```

> Anschließend wird die Evaluierung durchgeführt.

Reviewer: Vielleicht
```suggestion
Anschließend wird eine Evaluierung durchgeführt.
```

> Über eine gewisse Zeit wird das Projekt dann im Rahmen eines Feldversuchs verfolgt und es wird beobachtet, in wie weit sich die Bewertung des untersuchten Projektes verändert.

Reviewer: Vielleicht ist das hier schon zu ausführlich?? Und: beobachten ist nur mit den Augen! 
Vielleicht
```suggestion
Über eine gewisse Zeit wird das Projekt dann im Rahmen eines Feldversuchs verfolgt und es wird anhand der ausgewählten Metriken ermittelt, in wie weit sich die Bewertung des untersuchten Projektes verändert.
```
Satz 90 fällt dann weg
statt anhand evtl. mit Hilfe der ausgewählten Metriken oder kann das an dieser Stelle evtl. sogar ganz weg, selbst ich weiß jetzt, wie es geht dann hieße es
```suggestion
Über eine gewisse Zeit wird das Projekt dann im Rahmen eines Feldversuchs verfolgt und es wird ermittelt, in wie weit sich die Bewertung des untersuchten Projektes verändert.
```

\maketitle

\section{Motivation}
Laut \cite{Balzert2009} entfallen 80\% des Aufwands in der Softwareentwicklung auf deren Wartung, wobei 40\% davon benötigt werden, um die zu wartende Software zu verstehen.
Ein erheblicher Teil der Kosten für die Softwareentwicklung entfällt damit auf das Verstehen der Software.
Wenn insgesamt die Entwicklungskosten reduziert werden sollen, stellt dieser Umstand einen wesentlichen Hebel dar.

> Denn der Wartungsaufwand im Übrigen lässt sich vielfach nicht reduzieren, weil sich die Anforderungen an die Software im Laufe der Zeit ändern, weil auftretende Probleme behoben werden müssen oder weil Abhängigkeiten auf andere Software, Frameworks oder Komponenten angepasst werden müssen.

Reviewer: Abhängigkeiten müssen auf etwas angepasst werden
Abhängigkeiten müssen an etwas angepasst werden?
oder
Abhängigkeiten von etwas     müssen angepasst werden?
Kann man eine Abhängigkeit anpassen?
Ich weiß es leider nicht

Reviewer: in 35 36 und 37 reduzieren!
Vorschlag:
in 35
Denn der Wartungsaufwand im Übrigen lässt sich vielfach kaum verringern, weil....

Reviewer: Denn der Wartungsaufwand im Übrigen ist kaum zu verringern, weil....

lässt sich verringern - ist auch zu ähnlihc zum folgenden

Author: Abhängigkeiten können geupdated (bzw. angepasst) werden und es sind immer Abhängigkeiten auf etwas

> Denn der Wartungsaufwand im Übrigen lässt sich vielfach nicht reduzieren, weil sich die Anforderungen an die Software im Laufe der Zeit ändern, weil auftretende Probleme behoben werden müssen oder weil Abhängigkeiten auf andere Software, Frameworks oder Komponenten angepasst werden müssen.

Reviewer: ```suggestion
Denn der Wartungsaufwand im Übrigen ist vielfach kaum zu verringern, weil sich die Anforderungen an die Software im Laufe der Zeit ändern, weil auftretende Probleme behoben werden müssen oder weil Abhängigkeiten auf andere Software, Frameworks oder Komponenten angepasst werden müssen.
```

Reviewer: lässt sich reduzieren kommt danach noch
das oben ist aber noch nicht die Endfassung für den zweiten Teil des Satzes

> Durch das erfüllen gewisser Qualitätsstandards lässt sich bereits der für das Verstehen des Codes notwendige Aufwand reduzieren.

Reviewer: ```suggestion
Jedoch lässt sich bereitts durch das Erfüllen gewisser Qualitätsstandards der für das Verstehen des Codes notwendige Aufwand reduzieren.
```

> Durch das erfüllen gewisser Qualitätsstandards lässt sich bereits der für das Verstehen des Codes notwendige Aufwand reduzieren.

Reviewer: ```suggestion
Bereits durch das Erfüllen gewisser Qualitätsstandards lässt sich jedoch der für das Verstehen des Codes notwendige Aufwand reduzieren.
```

Qualitativ hochwertige Software zeichnet sich nach \cite{ISO-25010} dadurch aus, dass sie stabil ist und gut gewartet, analysiert, erlernt und verändert werden kann.
Software, die diese Anforderungen erfüllt, beinhaltet Code, der nach bestimmten Richtlinien gebildet wird.
Voraussetzung ist jedoch, dass sich alle Entwickler eines Teams an diese vorher festgelegten und für die gesamte Projektlaufzeit geltenden Richtlinien halten.

> Zu diesen Regeln gehört beispielsweise, dass Code auf einer Ebene einheitlich eingerückt wird, dass öffnende geschweifte Klammern in der gleichen Zeile stehen, wie der Code, der die Klammer erfordert, dass Variablennamen sprechend sind und nicht nur aus einem Buchstaben bestehen oder dass Parameter auf ihre Gültigkeit überprüft werden, bevor mit ihnen gearbeitet wird.

Reviewer: das nach stehen muss weg, es folgt kein neuer satz!
```suggestion
Zu diesen Regeln gehört beispielsweise, dass Code auf einer Ebene einheitlich eingerückt wird, dass öffnende geschweifte Klammern in der gleichen Zeile stehen wie der Code, der die Klammer erfordert, dass Variablennamen sprechend sind und nicht nur aus einem Buchstaben bestehen oder dass Parameter auf ihre Gültigkeit überprüft werden, bevor mit ihnen gearbeitet wird.
```

Reviewer: das komma nach stehen (Komma ist verloren gegagen

Reviewer: das wort komma ist beim kommentar verloren gegangen

Die Einhaltung solcher Regeln scheint auf den ersten Blick nur von geringer Bedeutung zu sein, allerdings lenken beispielsweise Unstimmigkeiten in der Formatierung beim Lesen des Codes ab und sie erschweren es, sich auf den eigentlichen Code zu konzentrieren.
In \cite{PJ2015} und \cite{SP2011} werden diese Formatverstöße als Hintergrundrauschen beschrieben, das vom eigentlichen Code ablenkt. 

\section{Problemstellung}
Code, der nach den oben beschriebenen Regeln geschrieben wurde, ist laut \cite{PJ2015} leichter verständlich und dadurch auch leichter erlernbar und veränderbar.

> Um die Teammitglieder dazu zu bringen, sich an diese Regeln zu halten, gibt es zwei Möglichkeiten.

Reviewer: ```suggestion
Um die Teammitglieder dazu zu bringen, sich an die vereinbarten Regeln zu halten, gibt es zwei Möglichkeiten.
```

> Zum einen kann im Reviewprozess des Projekts ein automatischer Test eingebaut werden, der dafür sorgt, dass grundsätzlich funktionierender Code dann nicht angenommen wird, wenn er gegen festlegte Regeln verstößt, wenn er also beispielsweise gegen Stylingrichtlinien verstößt oder es potentiell unsichere Variablenzugriffe gibt.

Reviewer: ```suggestion
Zum einen kann im Reviewprozess des Projekts ein automatischer Test eingebaut werden, der dafür sorgt, dass grundsätzlich funktionierender Code dann nicht angenommen wird, wenn er gegen festgelegte Regeln verstößt, wenn er also beispielsweise Stylingrichtlinien nicht einhält oder es potentiell unsichere Variablenzugriffe gibt.
```

> Zum anderen können die Teammitglieder durch Belohnungen dazu motiviert werden, von sich aus dafür zu sorgen, dass sie die festgelegten Regeln einhalten und dies auch selbst überprüfen. 

Reviewer: das dafür zu sorgen ist mE überflüssig oder siehst du das anders?
```suggestion
Zum anderen können die Teammitglieder durch Belohnungen dazu motiviert werden, von sich aus die festgelegten Regeln einzuhalten und dies auch selbst zu überprüfen. 
```

Um eine solche Motivation zu erreichen, kann Gamification verwendet werden.
Hierbei werden Methoden genutzt, die aus Computerspielen bekannt sind.

> Dazu gehören beispielsweise Levels, Badges oder Leader Boards, die das Team oder das einzelne Teammitglied erreichen kann. 

Reviewer: ```suggestion
Dazu gehören beispielsweise Levels, Badges oder Leader Boards, die das Team oder ein einzelnes Teammitglied erreichen kann. 
```

> Hierzu gehören beispielsweise Levels, Badges oder Leader Boards, die das Team oder das einzelne Teammitglied erreichen kann. 

Reviewer: statt hierzu Dazu oder

Reviewer: oder entfällt

> Beide Verfahren führen dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen, entweder, weil sie gelernt haben, dass der eingereichte Code sonst aus Formgründen abgelehnt wird oder weil sie für besseren Code beispielsweise mehr Erfahrungspunkte bekommen.

Reviewer: ```suggestion
Beide Verfahren führen dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen, entweder weil sie gelernt haben, dass der eingereichte Code sonst aus Formgründen abgelehnt wird oder weil sie für besseren Code beispielsweise mehr Erfahrungspunkte bekommen.
```

> Beide Verfahren führen dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen, entweder, weil sie gelernt haben, dass der eingereichte Code sonst aus Formgründen abgelehnt wird oder weil sie für besseren Code beispielsweise mehr Erfahrungspunkte bekommen.

Reviewer: Ganz wichtige Frage:
Hier schreibst du es so, als ob du schon weißt, das Gamification erfolgreich ist. Soll das nicht mehr so die These für die Arbeit sein????

Reviewer: oder weißt du das schon aus siehe zeile 71 - aber was ist dann deine genaue These?? / was deine Arbeit von den anderen unterscheidet? Die Art der Gaming-Elemente...???

> Beide Verfahren führen dazu, dass die Teammitglieder diese Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen, entweder, weil sie gelernt haben, dass der eingereichte Code sonst aus Formgründen abgelehnt wird oder weil sie für besseren Code beispielsweise mehr Erfahrungspunkte bekommen.

Reviewer: ```suggestion
Beide Verfahren führen dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen, entweder, weil sie gelernt haben, dass der eingereichte Code sonst aus Formgründen abgelehnt wird oder weil sie für besseren Code beispielsweise mehr Erfahrungspunkte bekommen.
```
die vereinbarten .... oder auch die Progr..., für diese fehlt der direktte Bezug

> Es entsteht also ein Lerneffekt, der bewirkt, dass die Teilnehmer auch in Zukunft qualitativ hochwertigen Code einreichen.

Reviewer: ```suggestion
Es ergibt sich also ein Lerneffekt, der bewirkt, dass die Teilnehmer auch in Zukunft qualitativ hochwertigen Code einreichen.
```

In beiden Fällen wird die eigene Motivation, guten Code zu schreiben, durch eine extrinsische Motivation verstärkt.
Im Rahmen dieser Arbeit liegt der Fokus auf der Motivation durch Gamification.

\section{Zielsetzung}
Ziel dieser Arbeit ist die Erweiterung der Codeanalyseplattform coderadar um Gamification-Elemente zur Steigerung der Code-Qualität.
Dazu werden zunächst ausgewählte Gamification-Elemente in coderadar implementiert.
Anschließend wird eine Balancierungsmatrix für eine gleichmäßige Verteilung der Punkte, auf denen Levels, Leader Boards und ähnliches basieren, erarbeitet.
Abschließend wird ein Feldversuch durchgeführt, um festzustellen, in wie weit die vorgestellten Maßnahmen die Softwarequalität nachhaltig verbessern konnten.

\section{Vorgehensweise}
Zunächst werden die Grundlagen dieser Arbeit vorgestellt.

> Dazu gehört die Erläuterung, was Gamification ist und in wie weit Gamification-Elemente in dieser Arbeit Verwendung finden.

Reviewer: ```suggestion
Es wird erläutert, was Gamification ist und welche Gamification-Elemente in dieser Arbeit Verwendung finden werden.
```
in wie weit ist mE komisch mir scheint welche besser...
finden werden: Zeitfrage: Bei den Sachen die du programmieren wirst, Futur? Futur hast du in 72 auch

Reviewer: aber weiter unten dann auch wieder nicht - denk mal drüber nach, wie es für dich am besten klingt.
sonst streiche in 70 das werden

Daneben wird im Rahmen der Vorstellung verwandter Arbeiten darauf eingegangen, welche Implementierungen und Implementierungsansätze es bereits für Gamification-Elemente in Zusammenhang mit der Code-Qualität gibt.
Außerdem wird die Codeanalyseplattform coderadar vorgestellt, die als Basis für die im Rahmen dieser Arbeit erfolgende Implementierung dienen wird.

> Im Anschluss daran wird die Einbindung von Gamification-Elementen vorgestellt.

Reviewer: in 69 war schon wird vorgestellt, passt hier wird erläutert?

> Dazu gehört zunächst die Auswahl der Metriken, anhand derer die Codequalität bewertet wird.

Reviewer: die Codequalität bewertet werden soll ??

> Entsprechend diesen Metriken wird der konkrete Code dann untersucht.

Reviewer: Ich hatte den Absatz überhaupt nicht verstanden.
ich glaube dieser Satz - und der nächste auch -  ist quatsch in diesem Zusammenhang....

Reviewer: du wolltest das Spielkonzept erläutern...

Author: Ich denke, es passt besser, wenn ich den Absatz einleite mit: Im Anschluss daran wird die Einbindung von Gamification-Elementen vorgestellt.

> Anhand des Ergebnisses dieser Untersuchung werden anschließend die Punkte verteilt, auf denen später Levels oder Leader Boards beruhen.

Reviewer: das Anhand des Ergebnisses dieser Untersuchung muss glaub ich anders

> Des Weiteren werden die Spielmechanismen, die Spielregeln und das Spielziel erläutert.

Reviewer: oh - jetzt schon wieder erläutert, passt hier werden erklärt?? oder beschrieben

> Außerdem wird eine Balancierungsmatrix ausgearbeitet, die für eine gleichmäßig gewichtete Punkteverteilung für unterschiedliche Aktionen sorgt.

Reviewer: ist das was anderes als in 77?

> Diese Balancierungsmatrix sorgt beispielsweise dafür, dass umfangreiche Änderungen verhältnismäßig mehr Punkte bringen als weniger umfangreiche Änderungen.

Reviewer: zwei mal sorgt für
```suggestion
Sie bewirkt beispielsweise, dass umfangreiche Änderungen verhältnismäßig mehr Punkte bringen als weniger umfangreiche Änderungen.
```

Danach wird auf die Implementierung eingegangen.

> Zunächst werden dafür die Anforderungen erhoben und eine Definition of Done, also die Definition, wann das Projekt fertig ist, erstellt.

Reviewer: ```suggestion
Zunächst werden die Anforderungen an die Implementierung erhoben und eine Definition of Done, also die Definition, wann das Projekt fertig ist, erstellt.
```
wenn man das so sagen kann

Auf Basis dieser Anforderungen werden dann die erforderlichen Änderungen an der coderadar-Plattform modelliert.

> Dann erfolgt die eigentliche Implementierung.

Reviewer: Was hältst du von
```suggestion
Es folgt die eigentliche Implementierung.
```

> Anschließend erfolgt die Evaluierung.

Reviewer: ```suggestion
Anschließend wird die Evaluierung durchgeführt.
```
Dann hast du nicht schon wieder folgt/erfolgt.

> Dazu wird zunächst ein Projekt zur Untersuchung in coderadar eingecheckt und anhand der ausgewählten Metriken wird die Code-Qulaität bewertet.

Reviewer: Da kannst du zwei Sätze draus machen, oder zumindest ein komma nach dem ersten Hauptsatz. Das erhöht die Lesbarkeit. beachte Qulaität
```suggestion
Dazu wird zunächst ein Projekt zur Untersuchung in coderadar eingecheckt, und anhand der ausgewählten Metriken wird die Code-Qualität bewertet.
```

Reviewer: ```suggestion
Dazu wird zunächst ein Projekt zur Untersuchung in coderadar eingecheckt, und anhand der ausgewählten Metriken wird dessen Code-Qualität bewertet.
```
falls dessen hier richtig ist

> Der hier ermittelte Wert soll als Grundlage für die Messung der Bewertungsänderung dienen.

Reviewer: ich würde da noch  evtl. ein künftige einfügen, ich weiß nur nicht ob:
für die künftige Messung der Bewertungsänderung
für die künftigen Messungen der Bewertungsänderungen
für die Messung/Messungen der künftigen Bewertungsänderung/en ???

> Über eine gewisse Zeit wird das Projekt dann im Rahmen eines Feldversuchs verfolgt und es wird beobachtet, in wie weit sich die Bewertung des untersuchten Projektes verändert.

Reviewer: musst du was hier dazu schreiben, wie du das "beobachten"/messen willst?

> Zusätzlich dazu sollen die Projektteammitglieder hinsichtlich der Arbeit mit den Gamification-Elementen in coderadar befragt werden.

Reviewer: ```suggestion
Zusätzlich dazu sollen die Projektteammitglieder zu ihrer Arbeit mit den Gamification-Elementen in coderadar befragt werden.
```

Abschließend wird das Ergebnis der Arbeit zusammengefasst.
Auf Basis der Bewertungsveränderung und der Ergebnisse der Umfrage wird ein Fazit zur vorgestellten Implementierung von Gamification-Elementen und ihrem Einfluss auf nachhaltige Code-Qualität gezogen.

> Als Grundlage für dieses Fazit dienen dabei zum Einen die Beobachtung über die Veränderung der Bewertung der Code-Qualität und zum Anderen die Ergebnisse aus der Befragung der Projektteammitglieder.

Reviewer: das steht schon in 94

Reviewer: wahrschienlich sollte 94 gestrichen und durch 95 ersetzt werden

> Als Grundlage für dieses Fazit dienen dabei zum Einen die Beobachtung über die Veränderung der Bewertung der Code-Qualität und zum Anderen die Ergebnisse aus der Befragung der Projektteammitglieder.

Reviewer: zum einen... zum anderen muss klein - habe ich nachgesehen. Das kam evtl. oben auch schon vor, hab jetzt keine Zeit das zu suchen.
Beobachtung oder Beobachtungen
muss die Bewertung da rein, reicht nicht
Beobachtun... über die Veränderung der Code-Qualität

\maketitle

\section{Motivation}
Laut \cite{Balzert2009} entfallen 80\% des Aufwands in der Softwareentwicklung auf deren Wartung, wobei 40\% davon benötigt werden, um die zu wartende Software zu verstehen.

> Zwangsläufig stellt somit das Verstehen der Software einen erheblichen Teil der Kosten der Softwareentwicklung dar. 

Reviewer: ```suggestion
Ein erheblicher Teil der Kosten für die Softwareentwicklung entfällt damit auf das Verstehen der Software.
```

Wenn insgesamt die Entwicklungskosten reduziert werden sollen, stellt dieser Umstand einen wesentlichen Hebel dar.

> Denn der Wartungsaufwand im Übrigen lässt sich vielfach nicht reduzieren, weil sich die Anforderungen an die Software im Laufe der Zeit ändern, weil auftretende Probleme behoben werden müssen oder weil Abhängigkeiten mit anderer Software oder Komponenten angepasst werden müssen.

Reviewer: Ist das inhaltlich das gleiche wie du geschrieben hast? 
Abhängigkeiten aktualisieren = Abhängigkeiten mit anderer Software oder Komponenten anpassen ?
(Kann man überhaupt sagen "Abhängigkeiten mit anderer Software oder Komponenten anpassen" das ist noch komisch Vielleicht eher
weil Abhängigkeiten zwischen....??? oder einzelnen Komponenten angepasst werden müssen.
weil Abhängigkeiten von anderer Software oder einzelnen Komponenten (dann geht nicht angepasst, vielleicht berücksichtigt werden müssen oder eher?? Muss jetzt erst aufhören

Reviewer: oder ist gemeint
weil Abhängigkeiten von/zwioschen/zu.(Präposition unklar) anderer Software berücksichtigt oder einzelne Komponenten angepasst werden müssen.
oder ist der Begriff Abhängigkeiten hier falsch

Author: Habe es angepasst.

> Aufwand für das Verstehen der Software lässt sich bereits dadurch reduzieren, dass der Code gewissen Qualitätsstandards entspricht, so dass das Verstehen der Software weniger aufwändig ist.

Reviewer: das mit dem Aufwand ist doppelt! muss jetzt aber erst aufhören

Author: Habe es angepasst. Der nächste Satz war auch noch eine Doppelung.

> Aufwand für das Verstehen der Software lässt sich bereits dadurch reduzieren, dass der Code gewissen Qualitätsstandards entspricht, so dass das Verstehen der Software weniger aufwändig ist.
> Dazu ist es notwendig, dass der Code gewisse Qualitätsstandards erfüllt.

Reviewer: ```suggestion
Der Aufwand für das Verstehen der Software lässt sich bereits dadurch reduzieren, dass der Code gewisse Qualitätsstandards erfüllt.
```

Author: Habe den Satz umformuliert.

> Qualitativ hochwertige Software zeichnet sich nach \cite{ISO-25010} dadurch aus, dass sie stabil ist und gut gewartet, analysiert, erlernt und verändert werden kann.

Reviewer: , dass sie stabil ist
oder 
dass sie stabil läuft
oder ein anderes Verb statt ist

Reviewer: oder ist ist Fachsprache?

Author: Das ist Fachspache, das geht so.

Um Code zu schreiben, der diese Anforderungen erfüllt, gibt es Richtlinien, an die sich alle Entwickler in einem Team halten sollten.
Diese Richtlinien besitzen zwar keine Allgemeingültigkeit, aber sie werden meist zu Projektbeginn festgelegt und gelten dann für die gesamte Projektlaufzeit.

> Zu diesen Regeln gehört beispielsweise, dass der Code auf einer Ebene einheitlich eingerückt wird, dass öffnende geschweifte Klammern in der gleichen Zeile sind, wie der Code, der die Klammer erfordert, dass Variablennamen sprechend sind und nicht nur aus einem Buchstaben bestehen oder dass Parameter auf ihre Gültigkeit überprüft werden, bevor mit ihnen gearbeitet wird.

Reviewer: ```suggestion
Zu diesen Regeln gehört beispielsweise, dass der Code auf einer Ebene einheitlich eingerückt wird, dass öffnende geschweifte Klammern in der gleichen Zeile stehen, wie der Code, der die Klammer erfordert, dass Variablennamen sprechend sind und nicht nur aus einem Buchstaben bestehen oder dass Parameter auf ihre Gültigkeit überprüft werden, bevor mit ihnen gearbeitet wird.
```

\section{Problemstellung}
Code, der nach den oben beschriebenen Regeln geschrieben wurde, ist laut \cite{PJ2015} leichter verständlich und dadurch auch leichter erlernbar und veränderbar.
Um die Mitglieder des Entwicklungsteams dazu zu bringen, sich an die vereinbarten Regeln zu halten, gibt es zwei Möglichkeiten.
Zum einen kann im Reviewprozess des Projekts ein automatischer Test eingebaut werden, der dafür sorgt, dass grundsätzlich funktionierender Code dann nicht angenommen wird, wenn er gegen festgelegte Regeln verstößt, wenn er also beispielsweise Stylingrichtlinien nicht einhält oder es potentiell unsichere Variablenzugriffe gibt.
Zum anderen können die Teammitglieder durch Belohnungen dazu motiviert werden, von sich aus die festgelegten Regeln einzuhalten und dies auch selbst zu überprüfen. 
Es stellt sich nun die Frage, ob eine solche Motivation durch Gamification erreicht werden kann.
Bei Gamification werden Methoden genutzt, die aus Computerspielen bekannt sind.
Dazu gehören beispielsweise Levels, Badges oder Leader Boards, die das Team oder ein einzelnes Teammitglied erreichen kann. 

> Die oben beschriebene Ablehnung von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt zwar dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.

Reviewer: der Aber-Satz fehlt, also zwar weg
```suggestion
Die oben beschriebene Ablehnung von eingereichtem Code auf Grund von Verstößen gegen Richtlinien führt dazu, dass die Teammitglieder die vereinbarten Programmierrichtlinien mit der Zeit von sich aus einhalten, ohne immer darauf hingewiesen werden zu müssen.
```

> Es ergibt sich also ein Lerneffekt, der bewirkt, dass die Teilnehmer auch in Zukunft qualitativ hochwertigen Code einreichen.

Reviewer: das auch ist mE überflüssig:
```suggestion
Es ergibt sich also ein Lerneffekt, der bewirkt, dass die Teilnehmer in Zukunft qualitativ hochwertigen Code einreichen.
```

\section{Zielsetzung}
Ziel dieser Arbeit ist die Erweiterung der Codeanalyseplattform coderadar um Gamification-Elemente zur Steigerung der Code-Qualität.
Dazu werden zunächst Gamification-Elemente ausgewählt und in coderadar implementiert.
Anschließend wird eine Balancierungsmatrix für eine gleichmäßige Verteilung der Punkte, auf denen Levels, Leader Boards und ähnliches basieren, erarbeitet.
Abschließend wird ein Feldversuch durchgeführt, um festzustellen, ob und in wie weit die vorgestellten Maßnahmen die Softwarequalität nachhaltig verbessern konnten.

\section{Vorgehensweise}
Zunächst werden die Grundlagen dieser Arbeit vorgestellt.
Es wird erläutert, was Gamification ist und welche Gamification-Elemente in dieser Arbeit Verwendung finden werden.
Daneben wird im Rahmen der Vorstellung verwandter Arbeiten darauf eingegangen, welche Implementierungen und Implementierungsansätze es bereits für Gamification-Elemente in Zusammenhang mit der Code-Qualität gibt.
Außerdem wird die Codeanalyseplattform coderadar vorgestellt, die als Basis für die im Rahmen dieser Arbeit erfolgende Implementierung dienen wird.

> Im Anschluss daran wird die Vorbereitungen für die Implementierung erläutert.

Reviewer: ```suggestion
Im Anschluss daran werden die Vorbereitungen für die Implementierung erläutert.
```

Dazu gehört zunächst die Auswahl der Metriken, anhand derer die Codequalität bewertet werden soll, und die Auswahl der Gamification-Elemente.

> Entsprechend der ausgewählten Metriken wird der Code später untersucht.

Reviewer: ```suggestion
Entsprechend den ausgewählten Metriken wird der Code später untersucht.
```

> Anhand der Ergebnisse dieser Untersuchung werden später die Punkte verteilt, auf denen dann die Gamification-Elemente beruhen.

Reviewer: ```suggestion
Anhand der Ergebnisse dieser Untersuchung werden dann die Punkte verteilt, die mit den Gamification-Elementen erreicht werden können.
```

Author: Ich finde das später an der Stelle besser, weil das ja nicht danach sondern erst in der Evaluierung geschieht. Der Zweite Teil des Satzes hat jetzt eine komplett andere Bedeutung. Die Gamification-Elemente helfen ja nicht direkt dabei, die Punkte zu erreichen, sondern sie dienen zur Visualisierung der Punkte bzw. ergeben sich aus den Punkten.

Reviewer: Das Problem ist, dass du zweimal später direkt hintereinander hast.

Reviewer: das mit der komplett anderen Bedeutung wussste ich nicht. Das entscheidest du am besten selbst.
für das zweite später:
passt vielleicht
```suggestion
Anhand der Ergebnisse dieser Untersuchung werden dann bei der Evaluierung die Punkte verteilt, auf denen schließlich die Gamification-Elemente beruhen.
```
(bei der Evaluierung oder in der Evaluierung??)
oder
```suggestion
Anhand der Ergebnisse dieser Untersuchung werden hinterher bei der Evaluierung die Punkte verteilt, auf denen dann die Gamification-Elemente beruhen.
```
oder statt hinterher danach? oder ganz ohne: werden bei der Evaluierung die Punkte verteilt, auf denen dann...
oder
mit Futur: 
```suggestion
Anhand der Ergebnisse dieser Untersuchung werden später die Punkte verteilt, auf denen dann die Gamification-Elemente beruhen.
```
```suggestion
Anhand der Ergebnisse dieser Untersuchung werden während des Feldversuchs die Punkte verteilt, auf denen dann die Gamification-Elemente beruhen.
```
oder: im Feldversuch

> Die Verteilung dieser Punkte erfolgt mit Hilfe einer Balancierungsmatrix, die für eine gleichmäßig gewichtete Punkteverteilung für unterschiedliche Aktionen sorgt.

Reviewer: ```suggestion
Um für unterschiedliche Aktionen die Punkte gleichmäßig gewichtet zu verteilen, wird eine Balancierungsmatrix definiert.
```

> Die Verteilung dieser Punkte erfolgt mit Hilfe einer Balancierungsmatrix, die für eine gleichmäßig gewichtete Punkteverteilung für unterschiedliche Aktionen sorgt.
> Diese Balancierungsmatrix wird ebenfalls an dieser Stelle definiert.

Reviewer: ```suggestion
Um für unterschiedliche Aktionen die Punkte gleichmäßig gewichtet zu verteilen, wird eine Balancierungsmatrix definiert.
```

Reviewer: Wie findest du das? Es war mir zu viel Punkte verteilen Verteilung der Punkte. Ist das sachlich noch richtig?
Da ist ein Satz weggefallen, weil ich zwei zu einem zusammengefasst habe. Ich hoffe, ich habe es richtig gemacht, guck es bitte nochmal nach.

Reviewer: oder muss es heißen
Um für unterschiedliche Aktionen die Punkte gleichmäßig gewichtet verteilen zu können, wird....definiert.

Author: Das mit zu können ist besser.
```suggestion
Um für unterschiedliche Aktionen die Punkte gleichmäßig gewichtet verteilen zu können, wird eine Balancierungsmatrix definiert.
```
